addons/website_sale_autocomplete/views/templates.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+
+<odoo>
+    <template id="address_autocomplete">
+        <div t-attf-class="dropdown-menu w-100 #{'show' if results else ''}">
+            <a class="dropdown-item js_autocomplete_result"
+                 t-foreach="results" t-as="result"
+                 t-att-data-google-place-id="result['google_place_id']">
+                <t t-out="result['formatted_address']"/>
+            </a>
+            <img class="p-2" src="/website_sale_autocomplete/static/src/img/powered_by_google_on_white.png" alt="Powered by Google"/>
+        </div>
+    </template>
+
+    <template id="website_sale_address_with_autocomplete" inherit_id="website_sale.address">
+        <xpath expr="//input[@name='street']" position="attributes">
+            <attribute name="t-att-data-autocomplete-enabled">
+                1 if website.google_places_api_key else 0
+            </attribute>
```

Comment:
FYI data-autocomplete-enabled is not set at all with this setup if the key is falsy.
Though it is good if we take the selector I proposed above.